Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.cs

```diff
@@ -203,6 +206,42 @@ private void EnsureRevocableReactivationRequestExistsFor(IdentityAddress activeI
             throw new DomainException(DomainErrors.NoRevocableReactivationRequestExists(activeIdentity));
     }
 
+    public void Decompose(IdentityAddress activeIdentity, DeviceId activeDevice)
+    {
+        EnsureRelationshipNotDecomposed(activeIdentity);
+        EnsureStatus(RelationshipStatus.Terminated);
+        EnsureHasParticipant(activeIdentity);
```

Comment:
I think it makes more sense to make this the first check

Reply:
Changes made.